### PR TITLE
feat: Dafny support

### PR DESCRIPTION
**Pull request by Gorgeous-Patrick**
Hi! This is the support for dafny. Tested on Linux

### CODE REVIEW

**Comment by Gorgeous-Patrick** [comment]
I am looking for a proper way to set the root dir though.

**Comment by glepnir** [comment]
you should use `vim.fn.win32 == 1` to check the system is windows or not. then change the cmd .

```lua
local bin_name = 'donat'
if vim.fn.has 'win32' == 1 then
  cmd = { 'cmd.exe', '/C', bin_name,.... }
end

```

File: lua/lspconfig/server_configurations/dafny.lua [hunk 1]
```diff
@@ -0,0 +1,20 @@
+local util = require 'lspconfig.util'
+
+local root_files = {
+    'compile_commands.json',
+    '.ccls',
+}
+return {
+  default_config = {
+    filetypes = { 'dfy', 'dafny'},
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/dafny.lua:9]:
```suggestion
    filetypes = { 'dfy', 'dafny' },
```

Comment (Gorgeous-Patrick) [reply lua/lspconfig/server_configurations/dafny.lua:9]: Ok I will do this a bit later today. But I currently don't have a windows machine. Probably I need help for that.

**Comment by Gorgeous-Patrick** [comment]
I think that I cannot make a 100% correct path here. Because the official documentation of the language server requires the users to compile themselves and put the compiled version ( a dll file or a exe file ) somewhere in their systems. I think there is no way to automatically find the file. 

The path written there is just the default path used in ArchLinux AUR package.

I will write the documentation to teach the users so that they can probably specify the path for themselves. IDK if that sounds good to you.

File: lua/lspconfig/server_configurations/dafny.lua [hunk 2]
```diff
@@ -0,0 +1,25 @@
+local util = require 'lspconfig.util'
+
+local bin_name = 'dotnet'
+local dll_path = '/opt/dafny/DafnyLanguageServer.dll'
+local cmd = {bin_name, dll_path}
+if vim.fn.has 'win32' == 1 then
+  dll_path = 'C:\\Program Files\\dafny\\DafnyLanguageServer.dll'
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/dafny.lua:7]: don't hardcode the path in there.

**Comment by Gorgeous-Patrick** [comment]
Resolved the issues. Probably good for another review @glepnir  Thanks.

File: lua/lspconfig/server_configurations/dafny.lua [hunk 3]
```diff
@@ -0,0 +1,20 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    filetypes = { 'dfy', 'dafny' },
+    root_dir = function(fname)
+      util.find_git_ancestor(fname)
+    end,
+    single_file_support = true
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/dafny.lua:9]:
```suggestion
    single_file_support = true,
```

**Comment by glepnir** [comment]
need fix this commit msg .

**Comment by Gorgeous-Patrick** [comment]
OK I changed the commit msg. Sorry I was too careless. Thank you! I think it is ready for another review. @glepnir

**Pull request merged**